### PR TITLE
Complete branch coverage for more files

### PR DESCRIPTION
**Pull request by corsonknowles**
Increase branch coverage minimum threshold.

Modernize `.rubocop_todo.yml` with Rubocop 1.6.8

In the further pursuit of full coverage for `Rubocop/RSpec`, this solves for additional files.

> [!NOTE]
> We remove a safe operator from a method used in a node matcher here, since it should always get a node.

Reconciled with:
* https://github.com/rubocop/rubocop-rspec/pull/1989

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

- [ ] Added the new cop to `config/default.yml`.
- [ ] The cop is configured as `Enabled: pending` in `config/default.yml`.
- [ ] The cop is configured as `Enabled: true` in `.rubocop.yml`.
- [ ] The cop documents examples of good and bad code.
- [ ] The tests assert both that bad code is reported and that good code is not reported.
- [ ] Set `VersionAdded: "<<next>>"` in `default/config.yml`.

If you have modified an existing cop's configuration options:

- [ ] Set `VersionChanged: "<<next>>"` in `config/default.yml`.

### CODE REVIEW

**Comment by bquorning** [comment]
Surprisingly little duplication between this and #1989 :yay:

**Comment by corsonknowles** [comment]
@bquorning @ydah I reconciled this with #1989 and it should be ready to go!

File: spec/rubocop/cop/rspec/stubbed_mock_spec.rb [hunk 1]
```diff
@@ -134,4 +134,23 @@
...
+    it 'returns "allow(subject)" for :is_expected' do
+      expect(cop.send(:replacement, :is_expected)).to eq('allow(subject)')
+    end
+
+    it 'returns "allow_any_instance_of" for :expect_any_instance_of' do
+      expect(cop.send(:replacement,
+                      :expect_any_instance_of)).to eq(:allow_any_instance_of)
+    end
+
+    it 'falls through silently and returns nil for unknown methods' do
```
Comment (corsonknowles) [inline spec/rubocop/cop/rspec/stubbed_mock_spec.rb:152]: technically this is all that's needed in here to solve for the `case` statement

**Comment by corsonknowles** [comment]
The Ruby head CI failure comes down to how rocket hashes are spaced in the parser.

**Comment by bquorning** [comment]
> The Ruby head CI failure comes down to how rocket hashes are spaced in the parser.

See https://github.com/rubocop/rubocop-rspec/blob/00b6c0387652f8f5140465f92b3299a746e9d18d/spec/rubocop/rspec/hook_spec.rb#L77-L83 for inspiration on how to work around that.

**Review by bquorning — APPROVED** [review]
Thanks

**Pull request merged**